### PR TITLE
Adding telemetry to the PHP client

### PR DESCRIPTION
**Pull request by hush-hush**
Adding Dogstatsd telemetry to the PHP client. Since batch metrics is not enable by default like other clients the telemetry is disable by default for the basic client.

### CODE REVIEW

File: tests/UnitTests/DogStatsd/SocketsTest.php [hunk 1]
```diff
@@ -999,43 +1049,49 @@ public function testGlobalTags()
...
-        $expectedUdpMessage = 'metric:42|ms|#my_tag:tag_value,dd.internal.entity_id:04652bb7-19b7-11e9-9cc6-42010a9c016d';
-        $argsPassedToSocketSendTo = $spy->argsFromSocketSendtoCalls[0];
-
-        $this->assertSame(
-            $expectedUdpMessage,
-            $argsPassedToSocketSendTo[1]
-        );
-        putenv("DD_ENTITY_ID");
-    }
+#    public function testGlobalTagsWithEntityIdFromEnvVar()
```
Comment (olivielpeau) [inline tests/UnitTests/DogStatsd/SocketsTest.php]: why do you have to disable this test? 😢

File: tests/UnitTests/DogStatsd/SocketsTest.php [hunk 2]
```diff
@@ -26,11 +26,49 @@ static function getPrivate($object, $property) {
         return $reflector->getValue($object);
     }
 
+    private function get_default(&$var, $default=null) {
+      return isset($var) ? $var : $default;
+    }
+
+    public function assertSameTelemetry($expected, $actual, $message="", $params = array())
```
Comment (olivielpeau) [inline tests/UnitTests/DogStatsd/SocketsTest.php]: nit: I was a bit confused by the name at first, maybe:
```suggestion
    public function assertSameWithTelemetry($expected, $actual, $message="", $params = array())
```

is better?

File: src/DogStatsd.php [hunk 3]
```diff
@@ -296,6 +336,7 @@ private function normalize_tags($tags)
      **/
     public function send($data, $sampleRate = 1.0, $tags = null)
     {
+        $this->metrics_sent += count($data);
```
[the remark this answers is not in the document]
Comment (hush-hush) [reply src/DogStatsd.php:339]: We actually count every metrics before sampling in all our client.

**Comment by hush-hush** [comment]
I don't think we need to update the batch size limit since it's arbitrary. The real fix would be to compute the batch byte size with the telemetry payload like we do in other client. This could be done in another PR.

**Pull request merged**